src/main/java/uk/gov/hmcts/reform/translate/service/DictionaryService.java

```diff
@@ -38,4 +52,46 @@ public Map<String, String> getDictionaryContents() {
 
         return Collections.emptyMap();
     }
+
+
+    public void putDictionary(final DictionaryRequest dictionaryRequest) {
+
+        val currentUser = securityUtils.getUserInfo();
```

Comment:
Since we have the user at this point, it would be a lot more efficient to determine which of the methods we need to use for the phrases beforehand, instead of in the loop. If the user has the `manage-translations` role, then we can process all of the phrases in the same way because the role can't change mid-loop.

Reply:
SI SI  very good comment . DONE